File: proxy-nohup.pl
#!/usr/bin/env perl

use Getopt::Long;
Getopt::Long::GetOptions(
    'help'    => \$wants_help,
    'version' => \$version,
    'exec=s'  => \$exec,
);
if ( defined($wants_help) ) {
    print "usage:\r\n \tproxy-nohup commandline!\n";
}
elsif ( defined($version) ) {
    print "0.0.1\n";
}
elsif ( defined($exec) ) {
    start($exec);
}

sub start {
    my $exec = shift;
    system("nohup $exec  >/dev/null 2>&1 &");
}

File: nodejs/proxy-nohup.js
#!/usr/bin/env node
var exec = require('child_process').exec;
var argv = require('minimist')(process.argv.slice(2));

function start(argv) {
	if (argv.exec) {
		console.log('nohup ' + argv.exec + ' >/dev/null 2>&1 &');
		exec('nohup ' + argv.exec + ' >/dev/null 2>&1 &',
			function (error, stdout, stderr) {
				if (error !== null) {
					console.log('exec error: ' + error);
				}
			});
	}else{
		console.log('usage:\r\n' +
				'\tproxy-nohup --exec commandline');
	}

}

start(argv);


